kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -102,6 +101,26 @@ class SparkProcessBuilder(
     }
   }
 
+  private lazy val sparkCoreScalaVersion: String = {
+    val pattern = Pattern.compile("^spark-core_(\\d+\\.\\d+).*.jar$")
+    Paths.get(sparkHome, "jars").toFile
+      .list((_, name) => pattern.matcher(name).matches())
+      .map { name => pattern.matcher(name).group(1) }
+      .head
+  }
+
+  override protected def engineScalaBinaryVersion: String =
+    StringUtils.defaultIfBlank(System.getenv("SPARK_SCALA_VERSION"), sparkCoreScalaVersion)
+
+  override protected lazy val engineHomeDirFilter: FileFilter = (file: File) => {
+    val pattern = if (SemanticVersion(SCALA_COMPILE_VERSION) >= "2.13") {
```

Comment:
equals should be enough, as we don't know what will happen for Scala 3 at the current stage
```suggestion
    val pattern = if (SemanticVersion(SCALA_COMPILE_VERSION) == "2.13") {
```

Reply:
SGTM.